packages/core-flows/src/product/steps/update-products.ts

```diff
@@ -16,6 +23,22 @@ export const updateProductsStep = createStep(
       ModuleRegistrationName.PRODUCT
     )
 
+    if ("products" in data) {
+      if (data.products.some((p) => !p.id)) {
+        throw new MedusaError(
+          MedusaError.Types.INVALID_DATA,
+          "Product id is required when doing a batch update"
```

Comment:
suggestion, very-nitpicky(non-blocking): The step is not aware of its consuming operation, so I would rename the error message to reflect that:
```suggestion
          "Product ID is required in input 'products'"
```

Reply:
Sounds good, and I agree